### ISSUE TITLE
R Forecast: One predictor class for each R model.

### ISSUE THREAD

**Issue by jaheba** [awslabs/gluon-ts#2770]
Currently, there is a single `Predictor` class for all (univariate) r models:

https://github.com/awslabs/gluonts/blob/a973a04d1681dcb3074dabb559a002ed0333d2f2/src/gluonts/ext/r_forecast/_univariate_predictor.py#L37

One can select different models by setting the `method_name` parameter.

Ideally, each model becomes its own class similar to how we wrap `statsforecast`.

This would also allow us to add some custom behaviour to some models.

**Comment by lostella** [comment]
I think this makes sense. Doesn't really need to be backwards incompatible, right?

**Comment by jaheba** [comment]
No, this would be breaking. Also, the interface wouldn't be that much different, so I think this is fine.

**Comment by melopeo** [comment]
For `statsforecasts` we have a class per method, for instance:
1. `ThetaPredictor(StatsForecastPredictor)`
2. `WindowAveragePredictor(StatsForecastPredictor)`

A bit of a disadvantage is that if `statsforecasts` decides to add more models, then we will have to manually add it to the wrapper. I am not sure if this is the kind of situation where we want to be.

**Comment by jaheba** [comment]
The number of models that we support is limited and each new model would require some code changes in any case.

**Comment by melopeo** [comment]
This is an assumption that can be discussed. I am not sure we should assume this. Although I agree that in terms of usability it would at a great amount of readability.

**Comment by jaheba** [comment]
> This is an assumption that can be discussed. I am not sure we should assume this. Although I agree that in terms of usability it would at a great amount of readability.


Not sure what you want to discuss about this:

https://github.com/awslabs/gluonts/blob/a973a04d1681dcb3074dabb559a002ed0333d2f2/src/gluonts/ext/r_forecast/R/univariate_forecast_methods.R#L54-L120
